### ISSUE TITLE
Audio Recorder: Impossible to set bitrate when recording directly to a file

### ISSUE THREAD

**Issue by yanivshaked** [Canardoux/tau#12]
When using the recorder to record to a stream, I can set the samples quality (bit rate, bits per sample, etc.) easily:
```
      _audioRecorder = TauRecorder();

      _dataFlow = new StreamController<List<int>>();
      chunkedStreamReader = new ChunkedStreamReader<int>(_dataFlow!.stream);
      _foodController = StreamController<TauFood>();
      _foodControllerSubscription = _foodController!.stream.listen(_handleIncomingFood, onError: _onFoodError, onDone: _onFoodDone);

      await _audioRecorder!.open(
        from: InputDeviceNode.mic(),
        to: OutputStreamNode(_foodController!.sink,
            codec: Pcm(
              AudioFormat.raw,
              sampleRate: AudioConstants.SampleRate,
              depth: Depth.int16,
              endianness: Endianness.littleEndian,
              nbChannels: NbChannels.mono,
            )),
      );

      await _audioRecorder!.record();
```

However, when using the recorder to record directly to a file, quality cannot be set - and the quality result is pretty poor:
```
      _audioRecorder = TauRecorder();
      await _audioRecorder!.open(
        from: InputDeviceNode.defaultSource(),
        to: OutputFileNode(_audioPath!, codec: Aac(AudioFormat.adts)),
      );
      await _audioRecorder!.record();
```

Please advise, is there something I am missing here or is this feature not ready yet?

**Comment by Larpoux** [comment]
Hi Yaniv
> or is this feature not ready yet

TL;DR : No, it's not ready yet

**Long version :**

In Flutter Sound V4, V5 et V6, the codec was just an enum.
But each codec can have different parameters.
For example : we can specify the endianness for PCM.
Also, if one day we support the MP3 encoding, we must be able to specify VBR (Variable Bit Rate) or FBR (Fixed Bit Rate).

It was not clean to add all those codec parameters during `StartRecorder()`. This is the reason why we have different classes for Codecs in V9. Like this it will be possible to specify the codecs parameters during the constructor.

But this is not finished : the parameters "endianness" and "depth" in the PCM codec are actually ignored.
And parameters for Aac are missing (at least the bitrate). This must be implemented before releasing the final 9.0.

But the real problem is always same : we are not enough developers on τ.
We are two contributors : @Alvarocda and me.
Alvaro is actually busy rewriting the UI Widget.
And me, I actually do not work on τ.
My wish is that someone else can work on τ and become the main developer instead of me.
